### DISCUSSION TITLE
How to install a static library version using vpckg manifest files?

### DISCUSSION THREAD

**Discussion by netcorefan1** [microsoft/vcpkg#30641] [Q&A] [answered]
Hello,
I'm trying to switch into manifest mode, but I got stuck in understanding how can be installed a static library version, just like we does with `vcpkg install lib:x64-windows-static`. By looking [here](https://stackoverflow.com/questions/65756232/how-to-install-a-static-library-version-using-vpckg-manifest-files) it looks like this is done through expressions, but the posted links are broken and I can't figure out why it doesn't work:
```
"dependencies": [
      "lib1",
      "lib2",
      {
        "name": "lib3",
        "platform" : "(windows & x64 & static)"
      }
]
```
lib1 and lib2 are installed while lib3 is ignored with no apparent error.
I can force static install with `-DVCPKG_TARGET_TRIPLET:STRING=x64-windows-static`, but this is applied to each dependency rather then lib3 only.
Thanks

**Comment by autoantwort** [comment, marked as answer]
You have to create your own triplet and use [per port customization](https://learn.microsoft.com/en-us/vcpkg/users/triplets#per-port-customization) to set the linkage for that port to static.

**Reply by netcorefan1** [reply]
Many thanks for clearing up my confusion. The link you posted contains all the info I need to achieve the goal.

@TheMaverickProgrammer does not seems happy with the response and I can understand his frustration. I also hope that in the future it will be easier like doing `"dependencies": [ "lib1:x64-windows-static" ]` rather then recurring to custom ports. For now, from my side, what's really matter is that it can be done.

**Reply by aiyolo** [reply]
> You have to create your own triplet and use [per port customization](https://learn.microsoft.com/en-us/vcpkg/users/triplets#per-port-customization) to set the linkage for that port to static.

ugly syntax!
